lib/fetch.js

```diff
@@ -125,7 +125,7 @@ var fetchGit = function (state, callback) {
             }
         },
         function (next) {
-            if (isDirectory(tmpPath + '/.git')) {
+            if (isDirectory(tmpPath + '/.git') && !state.options.keepHistory && !(state.flags && state.flags['keep-history'])) {
```

Comment:
Oh, huh.  Well I didn't actually need it from `~/.pollen`, I was just trying to mimic everything else, so that's fine with me.